src/file/multi.ts

```diff
@@ -85,12 +80,7 @@ export class LinkedMetaPage {
     const count = view.getUint8(POINTER_BYTES);
 
     if (this.index < count - 1) {
-      return new LinkedMetaPage(
-        this.resolver,
-        this.offset,
-        this.index + 1,
-        this.metaPageData,
-      );
+      return new LinkedMetaPage(this.resolver, this.offset, this.index + 1);
```

Comment:
The promise still needs to be passed here, right? Otherwise the cache won't persist across two slots' `LinkedMetaPage`s.

Reply:
Done. Thank you